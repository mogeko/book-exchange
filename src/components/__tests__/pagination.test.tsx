import { fireEvent, render, renderHook } from "@testing-library/react";
import "@testing-library/jest-dom";
import Pagination from "@/components/pagination";
import { useState } from "react";

describe("Pagination", () => {
  it("renders a Pagination", () => {
    const hookResult = renderHook(() => useState(0));
    const { container } = render(
      <Pagination
        index={hookResult.result.current[0]}
        setIndex={hookResult.result.current[1]}
        length={3}
      />
    );

    expect(container.querySelector("div")).toHaveClass("btn-group");
  });

  it("render a Pagination then lengh <= 1", () => {
    const hookResult = renderHook(() => useState(0));
    const { container } = render(
      <Pagination
        index={hookResult.result.current[0]}
        setIndex={hookResult.result.current[1]}
        length={0}
      />
    );

    expect(container.hasChildNodes()).not.toBeTruthy();
  });

  it("try cleck pages buttons", () => {
    const hookResult = renderHook(() => useState(0));
    const { container } = render(
      <Pagination
        index={hookResult.result.current[0]}
        setIndex={hookResult.result.current[1]}
        length={3}
      />
    );

    const buttons = container.querySelectorAll(".btn-group button");
    expect(buttons.length).toBe(5);
    expect(buttons[1]).toHaveClass("btn-active");
    fireEvent.click(buttons[2]);
    expect(hookResult.result.current[0]).toBe(1);
  });

  it("try cleck last/next buttons", () => {
    const hookResult = renderHook(() => useState(0));
    const { container } = render(
      <Pagination
        index={hookResult.result.current[0]}
        setIndex={hookResult.result.current[1]}
        length={3}
      />
    );

    const buttons = container.querySelectorAll(".btn-group button");
    expect(buttons?.[1]).toHaveClass("btn-active");
    fireEvent.click(buttons?.[0] as HTMLButtonElement);
    expect(hookResult.result.current[0]).toBe(0);
    fireEvent.click(buttons?.[4] as HTMLButtonElement);
    expect(hookResult.result.current[0]).toBe(1);
  });

  it("snapshot a Pagination", () => {
    const hookResult = renderHook(() => useState(0));
    const { container } = render(
      <Pagination
        index={hookResult.result.current[0]}
        setIndex={hookResult.result.current[1]}
        length={3}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
