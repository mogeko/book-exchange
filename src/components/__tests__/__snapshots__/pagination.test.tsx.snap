// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render a Pagination then lengh <= 1 1`] = `<div />`;

exports[`Pagination renders a Pagination 1`] = `
<div>
  <div
    class="btn-group mx-auto"
  >
    <button
      class="btn btn-xs"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        />
      </svg>
      <span
        class="sr-only"
      >
        Previous Page
      </span>
    </button>
    <button
      class="btn btn-xs btn-active"
      value="0"
    >
      1
    </button>
    <button
      class="btn btn-xs"
      value="1"
    >
      2
    </button>
    <button
      class="btn btn-xs"
      value="2"
    >
      3
    </button>
    <button
      class="btn btn-xs"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        />
      </svg>
      <span
        class="sr-only"
      >
        Next Page
      </span>
    </button>
  </div>
</div>
`;
